src/data.ts

```diff
@@ -401,23 +477,41 @@ export class CddaData {
   }
 
   all(): SupportedTypeMapped[] {
-    return this._raw;
+    return this._rawAll;
   }
 
   _flatten<T = any>(_obj: T): T {
     const obj: any = _obj;
     if (this._flattenCache.has(obj)) return this._flattenCache.get(obj);
-    const parent =
-      "copy-from" in obj
-        ? this._byTypeById.get(mapType(obj.type))?.get(obj["copy-from"]) ??
-          this._abstractsByType.get(mapType(obj.type))?.get(obj["copy-from"])
-        : null;
-    if ("copy-from" in obj && !parent)
-      console.error(
-        `Missing parent in ${
-          obj.id ?? obj.abstract ?? obj.result ?? JSON.stringify(obj)
-        }`
-      );
+
+    function allSources(o: any): any[] {
+      const sources: any[] = [];
+      sources.push(o.__self);
+      while (o.__prevSelf) {
+        sources.push(o.__prevSelf);
+        o = o.__prevSelf;
+      }
+      return sources.reverse();
+    }
+
+    let parent: any = null;
+    if (obj.__prevSelf != null) {
```

Comment:
just realised that this means we show the copy-from links in the UI even for non-mod items. that's a nice little upgrade!